src/donations.py

```
@@ -0,0 +1,99 @@
+
+def dollars_to_cents(amount_dollars) -> int:
+    """Convert a dollar amount to cents using exact decimal arithmetic."""
+    d = Decimal(str(amount_dollars))
+    return int((d * 100).to_integral_value(rounding=ROUND_HALF_UP))
+
+
+def cents_to_dollars(amount_cents: int) -> float:
+    return float(Decimal(amount_cents) / 100)
+
+
+def validate_donation_payload(body: dict) -> tuple[Optional[dict], Optional[str]]:
```

Ruff (0.16.0)

[warning] Too many return statements (9 > 6)

(PLR0911)